env/dev/secretsmanager.tf

```diff
@@ -0,0 +1,166 @@
+# create the secretsmanager secret
+resource "aws_secretsmanager_secret" "sm_secret" {
+  name = "${var.app}-${var.environment}"
+  tags = "${var.tags}"
+  policy = "${data.aws_iam_policy_document.sm_resource_policy_doc.json}"
+}
+
+# get the saml user info so we can get the unique_id
+data "aws_iam_role" "saml_role" {
+  name = "${var.saml_role}"
+}
+
+# resource policy doc that limits access to secret
+data "aws_iam_policy_document" "sm_resource_policy_doc" {
+  statement = {
+    sid = "DenyWriteToAllExceptSAMLUsers"
+    effect = "Deny"
+    principals = {
+      type = "AWS"
+      identifiers = ["*"]
+    }
+    actions = [
+      "secretsmanager:CancelRotateSecret",
+      "secretsmanager:CreateSecret",
+      "secretsmanager:DeleteSecret",
+      "secretsmanager:PutSecretValue",
+      "secretsmanager:RestoreSecret",
+      "secretsmanager:RotateSecret",
+      "secretsmanager:TagResource",
+      "secretsmanager:UntagResource",
+      "secretsmanager:UpdateSecret",
+      "secretsmanager:UpdateSecretVersionStage",
+    ]
+    resources = [
+      "arn:aws:secretsmanager:${var.region}:${data.aws_caller_identity.current.account_id}:secret:${var.app}-${var.environment}-*"
+    ]
+    condition = {
+      test = "StringNotLike"
+      variable = "aws:userId"
+      values = [
+        "${data.aws_caller_identity.current.user_id}",
+        "${data.aws_caller_identity.current.account_id}",
+        "${formatlist("%s:%s", data.aws_iam_role.saml_role.unique_id, var.saml_users)}"
+      ]
+    }
+  }
+
+  statement = {
+    sid = "DenyReadToAllExceptSAMLandApp"
+    effect = "Deny"
+    principals = {
+      type = "AWS"
+      identifiers = ["*"]
+    }
+    actions = [
+      "secretsmanager:DescribeSecret",
+      "secretsmanager:List*",
+      "secretsmanager:GetRandomPassword",
+      "secretsmanager:GetSecretValue",
+    ]
+    resources = [
+      "arn:aws:secretsmanager:${var.region}:${data.aws_caller_identity.current.account_id}:secret:${var.app}-${var.environment}-*"
+    ]
+    condition = {
+      test = "StringNotLike"
+      variable = "aws:userId"
+      values = [
+        "${aws_iam_role.app_role.unique_id}:*",
+        "${data.aws_caller_identity.current.user_id}",
+        "${data.aws_caller_identity.current.account_id}",
+        "${formatlist("%s:%s", data.aws_iam_role.saml_role.unique_id, var.saml_users)}"
+      ]
```

Comment:
Consider using [locals](https://www.terraform.io/docs/configuration/locals.html) for repeated expressions.